sklearn/tests/test_naive_bayes.py

```diff
@@ -438,6 +445,61 @@ def test_discretenb_coef_intercept_shape(cls):
     assert clf.intercept_.shape == (1,)
 
 
+@pytest.mark.parametrize("cls", [MultinomialNB, ComplementNB, BernoulliNB,
+                                 CategoricalNB])
+def test_discretenb_degenerate_one_class_case(cls):
+    # The first axis of most array attributes of a discrete naive Bayes
+    #     classifier should have one element per class. Exceptions include:
+    #     ComplementNB.feature_all_, CategoricalNB.n_categories_.
+    # Confirm that this is the case for binary problems and degenerate
+    #     one-class problems when fitting with `fit` or `partial_fit`.
+    # Non-regression test for handling degenerate one-class case.
+
+    attr_list = ['classes_', 'class_count_', 'class_log_prior_',
+                 'feature_count_', 'feature_log_prob_']
+    X = [[1, 0, 0], [0, 1, 0], [0, 0, 1]]
+    y = [1, 1, 2]
+
+    def test_first_axis_lengths(expected_first_axis_length):
+        # Test first axis lengths of array attributes of clf
+
+        for attr in attr_list:
+            if hasattr(clf, attr):  # CategoricalNB has no feature_count_
+                attribute = getattr(clf, attr)
+                if isinstance(attribute, np.ndarray):
+                    assert attribute.shape[0] == expected_first_axis_length
+                else:  # CategoricalNB.feature_log_prob_ is a list
+                    for element in attribute:
+                        assert element.shape[0] == expected_first_axis_length
+
+    def test_prediction():
+        assert clf.predict(X[:1]) == y[0]
+
+    # Binary problem, fit method
```

Comment:
I think that you can parametrize this part instead of repeating the test

Reply:
Thanks for your review! I may not get a chance to follow up on this point until after the holidays, but if not, I will do so shortly afterward.

Reply:
I've parametrized this test, and I agree that it's cleaner now: 313d74f.
Thanks for the suggestion, and thanks for your review!
I want to apologize in advance that I won't be able to respond again until after the holidays, but I'd be more than happy to address any additional critiques the first week of January.